### PR TITLE
Add composer build packages description

### PR DESCRIPTION
**Pull request by eSlider**
In order to get out from building packages with phing I created and described building processes over composer.
Please give me you review and check described package building commands.

@RobinSchwammborn, please test and improve documentation, if there some typos. Thanks!

### CODE REVIEW

**Comment by eSlider** [comment]
Here is an direct link to view change rendered as HTML: 
https://github.com/mapbender/mapbender-starter/blob/edf4723382ad41258bff5eadc5e08f19bb82c064/CONTRIBUTING.md#building-packages

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -592,6 +592,30 @@ git commit -m "Merge 'hotfix/bug-short-description'"
 git push
 ``` 
 
+# Building packages 
+
+For distributing and building packages there is an [composer] command:
+
+* `bin/composer build` Build package command with follow optional parameters:
+    * **[tar.gz|zip]** - Optional parameter defines package file format. Default configuration is defined in `composer.json` as `config/archive-format`.
```
Comment (AxxL) [inline CONTRIBUTING.md]: Ich würde hier noch den Unterschied zwischen soft- und harddlink  beschreiben, wenn es das Feature noch gibt. tar.gz = web/bundles als softlinks (linux), zip = web/bundles als hardlinks (Windows).

**Comment by debugteam** [comment]
I am a simple man, i see composer, i upvote...

Why use an extra deployment tool, when you can have a package manager which can do everything.

This doesn't necessarily collide with Unix Philosophy's "one job one tool" dogma.
You just have to define "package building and deployment is the same thing".

**Comment by eSlider** [comment]
@debugteam, thanks for your sarcasm, but take a look at some pakage manager documentations to compare they functionality and what they do?

* https://getcomposer.org/doc/04-schema.md#archive
* https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html
* https://docs.npmjs.com/cli/build

**Comment by astroidex** [comment]
1. The description does not work in my installation it always creates a package with the name mapbender3-starter-3.0.6.0.zip instead of test-distribution1.0.1.zip
` bin/composer build zip test-distribution 1.0.1`

2. I do not want symlinks. How can I parse that in a bin/composer command? It should be possible. This should be documented too.

**Comment by eSlider** [comment]
@astroidex,
1. Build with other name was buggy and fixed now.
2. If you create zip, then there is no symlinked files

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,11 +4,11 @@ The Mapbender team welcomes contributions from all members - so you are welcome
 
 Third-party patches are essential for the preservation of high standards in [Mapbender].
 
-We simply cannot access the huge number of platforms and myriad configurations for running [Mapbender]. 
+We simply cannot access the huge number of platforms and myriad configurations which run [Mapbender].
```
Comment (atd-schubert) [inline CONTRIBUTING.md]: I would prefer `that` instead of `which`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -4,11 +4,11 @@ The Mapbender team welcomes contributions from all members - so you are welcome
 
 Third-party patches are essential for the preservation of high standards in [Mapbender].
 
-We simply cannot access the huge number of platforms and myriad configurations for running [Mapbender]. 
+We simply cannot access the huge number of platforms and myriad configurations which run [Mapbender]. 
 
 We want it as easy as possible to carry out changes to get the [modules] in your environment to run. 
 
-There are a few [guidelines][rules] that we need contributors to follow so that we can have a chance of keeping on top of things.
+Therefore, we provide a few [guidelines][rules] that every contributor has to follow, in order that we have a chance of keeping on top of things.
```
Comment (atd-schubert) [inline CONTRIBUTING.md]: I would recommend:
`Therefore, we provide [contribution-guidelines][rules], in order to have a chance of keeping on top of things.`

What is the meaning of `keeping on top of things`? Maybe we can find a better wording.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -560,60 +560,84 @@ It's mandatory to use *release/* prefix in your branch name.
...
-* Push on [github]
+* Push on [github]:
 
 ```sh
 git push
 ``` 
 
+# Building packages 
+
+For distributing and building packages, there is a matching [composer] command:
```
Comment (atd-schubert) [inline CONTRIBUTING.md]: There are special [composer] commands for distributing and building packages:

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -560,60 +560,84 @@ It's mandatory to use *release/* prefix in your branch name.
...
+
+* `bin/composer build` Build package command with follow optional parameters:
```
Comment (atd-schubert) [inline CONTRIBUTING.md]: Command to build a package with the following optional parameters:

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -560,60 +560,84 @@ It's mandatory to use *release/* prefix in your branch name.
...
+    * **[tar.gz|zip]** - Optional parameter that defines the package file format. The default configuration is defined in `composer.json` as `config/archive-format`.
+    * **[dist-name]** - Optional parameter that defines the package file name prefix. The default configuration is defined in `composer.json` as `name`, a vendor name will be ignored.
+    * **[dist-version]** - Optional parameter that defines the package version. This is included as suffix in the package name. The default configuration is defined in `composer.json` as `version`.
+
+You can define the [composer] distributing path in `composer.json` as `config/archive-dir`. The default location is a `dist` folder located in root of the project.
```
Comment (atd-schubert) [inline CONTRIBUTING.md]: ...is the `dist` folder...

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -560,60 +560,84 @@ It's mandatory to use *release/* prefix in your branch name.
...
+* `bin/composer build` Build package command with follow optional parameters:
+    * **[tar.gz|zip]** - Optional parameter that defines the package file format. The default configuration is defined in `composer.json` as `config/archive-format`.
+    * **[dist-name]** - Optional parameter that defines the package file name prefix. The default configuration is defined in `composer.json` as `name`, a vendor name will be ignored.
+    * **[dist-version]** - Optional parameter that defines the package version. This is included as suffix in the package name. The default configuration is defined in `composer.json` as `version`.
+
+You can define the [composer] distributing path in `composer.json` as `config/archive-dir`. The default location is a `dist` folder located in root of the project. 
+
+## Build package example
+
+Running this command distributes and builds the package `dist/test-distribution.1.0.1.tar.gz`:
```
Comment (atd-schubert) [inline CONTRIBUTING.md]: You can build and distribute an artifact to `dist/test-distritution.1.0.1.tar.gz` by running:

File: CONTRIBUTING.md (hunk 3 above)
Comment (RobinSchwammborn) [inline CONTRIBUTING.md]: It means: den Überblick behalten
A more obvious wording would be: "keep/maintain the/an overview" ;-)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -487,71 +487,71 @@ It's mandatory to use the "feature/" prefix in the branch name.
...
-* Create [pull-request]
+* Create [pull-request]:
 
-Then please wait for the feedback. We will check it out and review your code to merge it in the branch.
+Then just wait for our feedback. We will check it out and review your code to merge it in the branch. Thanks!
 
 ## Bug fix branch
 
-It's mandatory to use "hotfix/" prefix in your branch name.
+It's mandatory to use the "hotfix/" prefix in your branch name.
```
Comment (LazerTiberius) [inline CONTRIBUTING.md]: Do we distinguish between bugfix and hotfix? I thought, hotfix means immediate new release with one fix and for bugfixes, we collect some and then release a new version.

Comment (AxxL) [reply CONTRIBUTING.md]: Yepp. Will change it. The small details are OK.

**Pull request merged**